docs/root/intro/arch_overview/http_connection_management.rst

```diff
@@ -111,6 +111,54 @@ previously attempted priorities.
 
 Envoy can be extended with custom retry plugins similar to how custom filters can be added.
 
+.. _arch_overview_internal_redirects:
+
+Internal redirects
+--------------------------
+
+Envoy supports handling 302 redirects internally, that is capturing a 302 redirect response,
+synthesizing a new request, sending it to the upstream specified by the new route match, and
+returning the redirected response as the response of the original request.
+
+Internal redirects are configured via the ref:`redirect action
+<envoy_api_field_route.RouteAction.redirect_action>` field in
+route configuration. When redirect handling is on, any 302 response from upstream is
+subject to the redirect being handled by Envoy.
+
+For a redirect to be handled successfully it must pass the following checks
+
+1. Be a 302 response
+2. Have a *location* header with a valid, fully qualified URL matching the scheme of the original request.
+3. The request must have been fully processed by Envoy.
+4. The request must not have a body
+5. The request must have not been previously redirected, as determined by the presence of an x-envoy-original-url header
+
+Any failure will result in redirect being passed downstream instead.
+
+Once the redirect has passed these checks, the request headers which were shipped to the original
+upstream will be modified by
+
+1. Putting the fully qualified original request URL in the x-envoy-original-url header
+2. Replacing the Authority/Host, Scheme, and Path headers with the values from the Location header
+
+The altered request headers will then have a new route selected, be sent through a new filter chain,
+and then shipped upstream with all of the normal Envoy request sanitization taking place. Note that
+HTTP connection manager sanitization such as clearing untrusted headers will only be applied once,
+per-route header modifications will be applied on both the original route and the second route, even
+if they are the same, so be mindful configuring header modification rules to avoid duplicating
```

Comment:
What would you suggest? I mean if you want to do redirects and add a Foo header I'd suggest having route rules which remove the foo header then add a foo header.  I don't know that we want to get into the business of trying to decide what the user wanted  because maybe they do want things added twice (internally we do service trace headers on internal redirect routes for example)

Reply:
Yeah I dunno. Maybe just pull this snippet out into a doc "attention" or "warning" so it stands out? Then we can see if people complain and/or are confused and reevaluate?

Reply:
Ping on this. Can you move this into a warning/attention block?